### PR TITLE
Signals

### PR DESCRIPTION
**Pull request by sheagcraig**
This PR adds a Django signal watcher that allows Sal users' profile level (i.e. RO, RW, or GA), staff, and active attributes to be updated based on group membership provided by the SAML assertion returned by a successful SAML auth during login.

Along with this change comes four new settings.py settings; group lists for each profile level, and a pref specifying the name to use to acces the SAML assertion's group membership attribute.

The signals.py will only take action on users who are members of one of the three configured types of groups. Local users or SAML users who are not members of any of the groups are left alone. This allows admins to create local Django users for testing, or to manually add SAML users to particular access levels who otherwise wouldn't have access.

This creates the responsibility to ensure that the SAML IdP does not allow users to login if they shouldn't have access. While this probably goes without saying, if you allow all SAML authed users access, any users who have no group membership in one of the three configured groups will get the default profile level and access. However, if any of these users _previously_ were members of any of the configured groups, they will retain that profile level and staff status, which the removal from the configured group was probably intended to revoke.

TL;DR use your SAML provider to limit access and the `SAML_READ_ONLY_GROUPS` setting to ensure that everyone who can login who shouldn't have admin rights gets properly handled.

### CODE REVIEW

**Comment by sheagcraig** [comment]
I would love some feedback about the behavior of the signals for users who have NO group membership.

I think this is correct, but I'd love some 👀

Alternate implementation would be to make the last block an `else` and anyone who isn't in a GA or RW group get "base" access (i.e. active, no staff, RO).

The problem with that approach is for when you are using sal-saml but are not using the groups feature. Then all of your manually-adjusted users are going to fall through to the `else` block and get downgraded to RO.

I suppose one way around that is to have it check; if all three group prefs are empty, assume the feature is disabled and do nothing.

Likewise, if my memory serves, the SO role basically doesn't do anything at all. So does it even make sense to have a `SAML_READ_ONLY_GROUPS` pref? Instead, would it make more sense to allow the SAML IdP to determine who is and isn't allowed to login, and give them the RO rights by default, unless they're in either RW or GA groups?

**Comment by grahamgilbert** [comment]
I’m deep into mdmdirector land at the movement so can’t test this, but your approach sounds sensible. I assume the use of this remains optional (forgive me if I missed this)

**Comment by weswhet** [comment]
I think this is the optimal way to treat this. 

“I suppose one way around that is to have it check; if all three group prefs are empty, assume the feature is disabled and do nothing.“

Otherwise looks solid. Thanks for working on this 🍻

**Comment by sheagcraig** [comment]
Yes; @grahamgilbert it's optional. If you're don't specify any groups, nothing happens. It only applies to people who are allowed to login by the IdP, when groups are configured, that the user is a member of. Everyone else is left alone.

**Review by grahamgilbert — APPROVED** [review]
